src/plib/library/DomainConnect.php

```diff
@@ -125,36 +181,92 @@ public function init()
             $resolvedIp = '';
         }
 
-        $hostingIps = $this->domain->getIpAddresses();
-
         if (in_array($resolvedIp, $hostingIps)) {
             \pm_Log::info("Domain {$this->domain->getDisplayName()} is already resolved to the current server");
 
             $this->domain->setSetting('connected', 1);
 
-            return;
+        } else {
+            $groupId[] = $groupIdWeb;
         }
 
         \pm_Log::debug("Domain {$this->domain->getDisplayName()} is resolved to {$resolvedIp}, but expected " . join(' or ', $hostingIps));
 
-        $serviceId = \pm_Config::get('webServiceId');
+//start configure MX and SPF records for mail service
+        try {
+            $resolvedIp = Dns::mxRecord($this->domain->getName());
+            $mxRecord = reset(array_filter($allRecords, function ($record) { return $record->type == 'MX'; }));
+            $mxRecordA = reset(array_filter($allRecords, function ($record) use ($mxRecord) { return $record->type == 'A' && $record->host == $mxRecord->pointsTo; }));
+            if ($resolvedIp['ip'] == $mxRecordA->pointsTo) {
+                \pm_Log::info("MX record for domain {$this->domain->getDisplayName()} is already resolved to the current server");
+                if ($this->domain->getSetting('connected', 1)) {
+                    return;
+                }
+            } else {
+                if ($this->domain->getSetting('connected', 1)) {
+                    $this->domain->setSetting('connected', 0);
+                }
+            }
+            $mxResolvedIp = ['ip' => $mxRecordA->pointsTo, 'host' => $mxRecord->pointsTo, 'priority' => $mxRecord->priority];
+            $groupId[] = $groupIdMail;
+        } catch (\pm_Exception $e) {
+            \pm_Log::warn($e);
+        }
+
+        try {
+            $hostname = gethostbyaddr(gethostbyname(gethostname()));
```

Comment:
better ask Plesk for hostname

Reply:
How?

Reply:
for example with API: https://github.com/plesk/api-php-lib/blob/master/src/Api/Struct/Server/GeneralInfo.php#L9

Reply:
Got it